pkg/cmd/checkendpoints/controller/pod_network_connectivity_check_controller.go

```diff
@@ -0,0 +1,108 @@
+package controller
+
+import (
+	"context"
+	"time"
+
+	operatorcontrolplanev1alpha1 "github.com/openshift/api/operatorcontrolplane/v1alpha1"
+	operatorcontrolplaneclientv1alpha1 "github.com/openshift/client-go/operatorcontrolplane/clientset/versioned/typed/operatorcontrolplane/v1alpha1"
+	alpha1 "github.com/openshift/client-go/operatorcontrolplane/informers/externalversions/operatorcontrolplane/v1alpha1"
+	"github.com/openshift/client-go/operatorcontrolplane/listers/operatorcontrolplane/v1alpha1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// PodNetworkConnectivityCheckController continuously performs network connectivity
+// checks and reports the results.
+type PodNetworkConnectivityCheckController interface {
+	factory.Controller
+}
+
+// controller implements a PodNetworkConnectivityCheckController that discovers the list of endpoints to
+// check by looking for PodNetworkConnectivityChecks in a given namespace, for a specific pod. Updates to
+// the PodNetworkConnectivityCheck status are queued up and handled asynchronously such that disruptions
+// to the ability to update the PodNetworkConnectivityCheck status do not disrupt the ability to perform
+// the connectivity checks.
+type controller struct {
+	factory.Controller
+	podName      string
+	podNamespace string
+	checksGetter operatorcontrolplaneclientv1alpha1.PodNetworkConnectivityCheckInterface
+	checkLister  v1alpha1.PodNetworkConnectivityCheckNamespaceLister
+	recorder     events.Recorder
+	// each PodNetworkConnectivityCheck gets its own ConnectionChecker
+	updaters map[string]ConnectionChecker
+}
+
+// Returns a new PodNetworkConnectivityCheckController that performs network connectivity checks
+// as specified in the PodNetworkConnectivityChecks defined in the specified namespace, for the specified pod.
+func NewPodNetworkConnectivityCheckController(podName, podNamespace string, checksGetter operatorcontrolplaneclientv1alpha1.PodNetworkConnectivityChecksGetter, checkInformer alpha1.PodNetworkConnectivityCheckInformer, recorder events.Recorder) PodNetworkConnectivityCheckController {
+	c := &controller{
+		podName:      podName,
+		podNamespace: podNamespace,
+		checksGetter: checksGetter.PodNetworkConnectivityChecks(podNamespace),
+		checkLister:  checkInformer.Lister().PodNetworkConnectivityChecks(podNamespace),
+		recorder:     recorder,
+		updaters:     map[string]ConnectionChecker{},
+	}
+	c.Controller = factory.New().
+		WithSync(c.Sync).
+		WithBareInformers(checkInformer.Informer()).
+		ResyncEvery(1*time.Second).
```

Comment:
the new structure means you only need to trigger on check informer updating and you only need to handle one time.

At the very least, make this a sync every minute so we don't burn cpu

Reply:
Done. Kept only sync on one minute. Didn't want to self trigger on status updates.